File: 07. Servicios y procesos/WorkSpace/EjerciciosCasa/src/ejercicioElectrodomesticos/Lavadora.java
package ejercicioElectrodomesticos;

public class Lavadora extends Electrodomestico {
	//Atributos
	private float carga; //Kilo
	private final float CARGA_DEFECTO = 5;
	
	//Constructores
	public Lavadora() {
		super();
		carga = CARGA_DEFECTO;
	}
	public Lavadora(float precio_base, float peso) {
		super(precio_base, peso);
		carga = CARGA_DEFECTO;
	}
	public Lavadora(float precio_base, String color, char consumo, float peso, float carga) {
		super(precio_base, color, consumo, peso);
		this.carga = carga;
	}
	
	//Métodos get
	public float getCarga() {
		return carga;
	}
	
	//Otros métodos
	public double precioFinal() {
		double preFinal = super.precioFinal();
		if(carga > 30) preFinal += 50;
		return preFinal;
	}
	
	@Override
	public String toString() {
		return "Lavadora [precio_base=" + precio_base + "€, color=" + color + ", consumo=" + consumo
				+ ", peso=" + peso + "kg, carga=" + carga + "kg]";
	}
	
	
	
}

File: 06. Python/POO_001/POO/POO_EJER_3/Lavadora.py
from POO.POO_EJER_3.Electrodomestico import Electrodomestico

class Lavadora(Electrodomestico):

    CARGA_DEFECTO = 5

    def __init__(self, precioBase=Electrodomestico.PRECIO_BASE_DEFECTO,
                 peso=Electrodomestico.PESO_DEFECTO,
                 color=Electrodomestico.COLOR_DEFECTO,
                 consumoEnergetico=Electrodomestico.CONSUMO_DEFECTO,
                 carga=CARGA_DEFECTO):

        super().__init__(precioBase, peso, color, consumoEnergetico)
        self.carga = carga

    def getCarga(self):
        return self.carga

    def precioFinal(self):

        precio = super().precioFinal()

        if self.carga > 30:
            precio += 50

        return precio

